### PR TITLE
Retry script for tests

### PR DESCRIPTION
**Pull request by md5**
As discussed in #1025, I've refactored the retry functionality into a helper script.

I've converted all of the tests with retry loops to use this function, except for `mongo-basics`. The reason I didn't convert that one yet is that it uses a different command for the retry loop and for the final attempt to get error output.

I also don't much like the use of `. "$dir/../../retry.sh"`, but I couldn't figure out a better way to do it. Besides the hokey relative path thing, the reason I source `retry.sh` instead of just executing it is to ensure that functions declared in the test are still in scope without having to use `export -f` to make them visible.

### CODE REVIEW

File: test/tests/jetty-hello-web/run.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+[ "$DEBUG" ] && set -x
```
Comment (yosifkit) [inline test/tests/jetty-hello-web/run.sh]: This looks unrelated :wink:, but I like it :smile:.

Comment (md5) [inline test/tests/jetty-hello-web/run.sh]: Ha! I usually use `git commit --verbose` to see what I'm committing, but it doesn't do any good if you don't actually look at the diff. Let me know if you want me to `--amend` it away.

Comment (yosifkit) [inline test/tests/jetty-hello-web/run.sh]: I'm ok with it.

**Comment by yosifkit** [comment]
LGTM

**Comment by md5** [comment]
ping @tianon

**Comment by tianon** [comment]
Nice, LGTM :+1:

(Sorry I'm so slow sometimes. :innocent:)

**Comment by md5** [comment]
:+1:

**Pull request merged**